### PR TITLE
add crud for groups

### PR DESCRIPTION
**Pull request by IhorSmulskyi**

### CODE REVIEW

File: app/views/teachers/groups/_form.html.haml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<%= form_for @group do |f| %>
```
Comment (mpanchuk) [inline app/views/teachers/groups/_form.html.haml]: `<%= form_for [:teachers, @group] do |f| %>`

File: app/views/teachers/groups/edit.html.haml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+%h1 Editing groups
+= form_with scope: :group, url: teachers_groups_path, local: true do |form|
```
Comment (mpanchuk) [inline app/views/teachers/groups/edit.html.haml]: `<%= form_for [:teachers, @group] do |f| %>`

File: app/controllers/teachers/groups_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  def new
+    @group = Group.new
+  end
+ 
+  def edit
+    @group = Group.find(params[:id])
+  end
+ 
+  def create
+    @group = Group.new params.require(:group).permit(:title)
```
Comment (Futursky) [inline app/controllers/teachers/groups_controller.rb:23]: `params.require(:group).permit(:title)` move this to separate method

File: app/controllers/teachers/groups_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    if @group.save
+      redirect_to  teachers_groups_path
+    else
+      render 'new'
+    end
+  end
+ 
+  def update
+ 
+    if @group.update params.require(:group).permit(:title)
```
Comment (Futursky) [inline app/controllers/teachers/groups_controller.rb]: same as above, you should create something like :group_params method

File: app/controllers/teachers/groups_controller.rb (hunk 3 above)
Comment (Futursky) [inline app/controllers/teachers/groups_controller.rb:23]: follow some code conventions, for example this https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md
`Group.new params.require(:group).permit(:title)`
`Group.find(params[:id])`
use round brackets always or don't use it at all

**Review by Futursky — CHANGES_REQUESTED** [review]
Overall looks ok, just two small comments.

**Pull request merged**